recipes/pugixml/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class PugiXmlConan(ConanFile):
+    name = "pugixml"
+    version = "1.10"
+    description = "Light-weight, simple and fast XML parser for C++ with XPath support"
+    topics = ("xml-parser", "xpath", "xml", "dom")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://pugixml.org/"
+    license = "MIT"
+    generators = "cmake"
+    settings = ("os", "arch", "compiler", "build_type")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "header_only": [True, False],
+        "wchar_mode": [True, False],
+        "no_exceptions": [True, False]
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'header_only': False,
+        'wchar_mode': False,
+        'no_exceptions': False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.header_only:
+            if self.settings.os != 'Windows':
+                del self.options.fPIC
+            del self.options.shared
+            self.settings.clear()
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_TESTS"] = False
+        if self.settings.os == 'Windows' and self.settings.compiler == 'Visual Studio':
+            cmake.definitions['CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS'] = self.options.shared
+        cmake.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder
+        )
+        return cmake
```

Comment:
Cache the cmake object to avoid reconfiguration in `package` after `build`.
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.definitions["BUILD_TESTS"] = False
        if self.settings.os == 'Windows' and self.settings.compiler == 'Visual Studio':
            self._cmake.definitions['CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS'] = self.options.shared
        self._cmake.configure(
            source_folder=self._source_subfolder,
            build_folder=self._build_subfolder
        )
        self._cmake cmake
```

Reply:
@madebr, thank you very much for suggested improvements!